linkerd/proxy/client-policy/src/lib.rs

```diff
@@ -469,12 +494,17 @@ pub mod proto {
                         Some(section)
                     };
 
+                    let port = u16::try_from(port)
+                        .ok()
```

Comment:
feels like this should return an `InvalidMeta` error if the port number is greater than 65535, rather than just having it be `None`?